Add ESP32-DevKitC evaluation boards

Soraxem opened the pull request:
##### SUMMARY
<!-- Describe the changes you have made. -->
I have added 5 out of 7 ESP32-DevKitC V4 evaluation-boards. I did not add two of those 7 because their datasheet recommended no new designs. I imagined using this for female pin headers, so the evaluation-board can be put in the socket. I added two layouts, one with lots of silkscreen prints and one without these, because for amateurs like me it is easier to look where the USB port is.

##### DATASHEETS / REFERENCES
<!-- When submitting a new device or package, please provide links to the
necessary datasheets, package drawings or standards documents.-->
[ESP32 DevKitC Getting Started](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/hw-reference/esp32/get-started-devkitc.html)
[ESP32-WROOM-32E / UE Datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-wroom-32e_esp32-wroom-32ue_datasheet_en.pdf)

##### OPEN QUESTIONS / UNRESOLVED ISSUES
<!-- List topics you are unsure about, if any. Otherwise, leave this empty. -->

##### CHECKLIST
<!-- Please review the following checklist and check the items you agree with
by adding an "x" between the brackets (i.e. "[x]"). If you are unsure, leave
the corresponding items unchecked so the library maintainers know where help
is needed. -->

- [X] I have read and followed the library conventions¹.
- [X] For packages, I followed IPC7351C (see details in library conventions).
- [X] For modified elements, I bumped their version number².
- [ ] I already used the new library elements in a design and verified their correctness (optional).
- [X] I'm the copyright owner of the added content (i.e. the changes are made by myself, not copied/imported from somewhere else).
- [X] I agree to publish all my changes under the CC0 Public Domain License³, allowing everyone to use and modify the content without any restrictions.

¹ *Library Conventions: https://docs.librepcb.org/#libraryconventions*
² *Minor version bump if only metadata was modified (e.g. "0.1" -> "0.1.1"), major version bump if functional changes were made (e.g. "0.1" -> "0.2")*
³ *CC0 Public Domain License: https://en.wikipedia.org/wiki/CC0*

Soraxem committed:
Added ESP32-DevKitC Symbol

Soraxem committed:
Added ESP32-DevKitC Component

Soraxem committed:
Added ESP32-DevKitC footprints for PCB and IPEX antenna

Soraxem committed:
Added Device ESP32-DevKitC-32UE

Soraxem committed:
Added all remaining ESP32-DevKitC devices (DA,VE,VIE)

Soraxem committed:
Used Different Package wit IPEX for ESP32-DevKitC-32UE

Soraxem committed:
Added Descriptions and Keywords to all ESP32-DevKitC devices

Soraxem committed:
Added Descriptions and Keywords to all ESP32-DevKitC packages

Soraxem committed:
Added second footprint variant

Added a footprint variant that is for permanent mounting and thus does not need any printed legends.

Soraxem committed:
Added second footprint variant

Added a footprint variant that is for permanent mounting and thus does not need any printed legends.

Soraxem committed:
Added manufacturer name

Added manufacturer name to packages

Soraxem committed:
Adjusted Courtyard

Soraxem commented:
Hello, is there any chance this PR will be reviewed anytime? Or is this in the current stage of development not feasable?

ubruhin commented:
Hi @Soraxem,

Sorry for the long delay! There's a lot of other work ongoing which makes it difficult to find time reviewing library additions...

Here my thoughts:

- Symbol/component: I'd highly prefer to make these more consistent with the already existing ESP32 library elements, in particular:
  - Rename pins/signals, e.g. "IO5" instead of just "5" and "IO7/D0" instead of just "D0". I think this makes it much more clear (avoid confusion with pad numbers).
  - Symbol pin arrangement - Same power/enable pin locations as the other ESP32 symbols (which is basically our standard for MCU-like symbols as it provides space for capacitors): `EN` - 5.08mm space - `5V` - 5.08mm space - `3V3` - 5.08mm space - `GND`. And I think I'd prefer having all I/Os on the right side, sorted by their IO number, without grouping :thinking: In the end, the grouping depends on the use-case so other people might prefer different grouping, while sorting by IO number is kind of neutral/universal...
- Packages:
  - I'm not sure about having two footprint variants for the silkscreen. What would you think about having one variant with something in between (outline of headers, board, ESP module and USB connector, but no more details)? Especially any texts other than `{{NAME}}` and `{{VALUE}}` should be removed anyway IMHO. And details like the buttons should be moved to the "Top Documentation" layer, to see them in assembly plans but not on silkscreen.
  - Not a big deal, but the button polygons are currently drawn as 4 independent lines, while a single polygon would be preferred.
  - The restring on the pads look a bit small to me, I suspect this could be difficult to solder. I'd suggest increasing the pad diameter to 1.7mm. Btw, do you actually know what drill diameter is used on the DevKitC board? I never found a detailed mechanical drawing specifying this...
  - There are library check warnings (no 3D model) which should be approved.

Would be nice if you could check/adjust these things :slightly_smiling_face:

Soraxem committed:
rearanging pins, and renaming them

Soraxem committed:
recreating DevKitC package

Soraxem committed:
adding square pad

Soraxem committed:
recreating DevKitC IPEX package

Soraxem committed:
renaming DevKitC signals

Soraxem commented:
I forgot this PR for a while now. But now i remembered.

- Symbol
  - Removed grouping, and listed them by gpio Number
  - Added Functionallity with slashes, i used this list from Espressiv: [ESP32 DevKitC Header Block](https://docs.espressif.com/projects/esp-dev-kits/en/latest/esp32/esp32-devkitc/user_guide.html#header-block)
  - I needed to shorten some functionalitys, so that the text fits onto the symbol
- Packages
  - Changed it to one footprint per Package
  - Removed additional text
  - Resized Pads, i do not know the actual drilling, but the header is the same as on the Arduino, so i now i just used the same dimensions as the Arduino uses.
  - Approved Library Checks

ubruhin commented:
Thanks!

- Symbol looks good :+1:
- Component looks good :+1: (just GND could be marked as a required signal, but it's ok without)
- The IPEX package is missing the documentation polygon around the package (the other package has it)

And I checked the different devices and realized in practice each of these 5 DevKits looks different (not just antenna vs. connector, also short vs. long ESP module, and the DA variant has a very special antenna shape). So actually we need 5 separate packages for them. I know this sounds a bit picky but this is the only way how we can provide accurate 3D models (once we add them some day). I could do the split if you like - no problem (easy thing with the "duplicate" feature in the library editor, just the device pinouts need to be redone). Just let me know.

Soraxem committed:
marking devkitc DA as deprecated

Soraxem committed:
adding DevKitC-32E specific package

Soraxem committed:
repurposing the DevKitC ipex package as the 32UE package

Soraxem committed:
adding DevKitC-VE specific package

Soraxem committed:
repurposing the DevKitC package as the VIE package

Soraxem committed:
approving missing 3d model

Soraxem commented:
I tired to fix the things highlighted. I did not add a package for the DA variant, because it is now marked EOL. I am unsure if it even shall be in the library, or if we just should remove it.

ubruhin committed:
Remove device "ESP32-DevKitC-DA"

ubruhin committed:
Add links to documentation

ubruhin commented:
Perfect, thank you!

> I am unsure if it even shall be in the library, or if we just should remove it.

Yeah I think that's better as it wouldn't have the correct package -> I removed it now.

And I added links to the DevKitC documentation to the component & devices.

ubruhin commented:
Btw, just for your information, generally we require legend drawings to be *outside* of the package outline and documentation polygons to be *inside* the package outlines. See other packages and the [library conventions](https://librepcb.org/docs/library-conventions/packages/#legend-layer).

However, in this case I'll accept as-is :slightly_smiling_face: 

Thanks again for these useful devices!
